src/app/contact/contact.component.ts

```diff
@@ -1,15 +1,54 @@
-import { Component, OnInit } from '@angular/core';
-
+import { Component, OnInit, Output, EventEmitter } from "@angular/core";
+import { FormGroup, FormControl, FormBuilder } from "@angular/forms";
+import { Validators } from "@angular/forms";
+import { CustomvalidationService } from "../services/customvalidation.service";
+import { User } from "../models/user";
 @Component({
-  selector: 'app-contact',
-  templateUrl: './contact.component.html',
-  styleUrls: ['./contact.component.scss']
+  selector: "app-contact",
+  templateUrl: "./contact.component.html",
+  styleUrls: ["./contact.component.scss"],
 })
 export class ContactComponent implements OnInit {
+  registerForm: FormGroup;
+  submitted = false;
+
+  @Output() userInfo = new EventEmitter<User>();
+
+  constructor(
+    private fb: FormBuilder,
+    private customValidator: CustomvalidationService
+  ) {}
 
-  constructor() { }
+  ngOnInit() {
+    this.registerForm = this.fb.group({
+      firstName: ["", Validators.required],
+      lastName: [""],
+      email: ["", [Validators.required, Validators.pattern("[^ @]*@[^ @]*")]],
```

Comment:
Consider also just using Angular's in-built email validator. I think you can add it here by adding `Validators.email` in your email form control